app/src/main/java/com/checkin/app/checkin/misc/views/OtpView.kt

```diff
@@ -0,0 +1,117 @@
+        etArray[0].requestFocus()
+    }
+
+    @BindViews(R.id.otp1, R.id.otp2, R.id.otp3, R.id.otp4, R.id.otp5, R.id.otp6)
+    internal lateinit var etArray: Array<EditText>
+
+    var otpValue: String
+        get() {
+            if (!validateOtp()) {
+                return ""
+            }
+            var otp = ""
+            for (et in etArray) {
+                otp += et.text.toString()
+            }
+            return otp
+        }
+        set(value) {
+            if (value.isBlank() && value.length != length) {
+                return
+            }
+            value.forEachIndexed { index: Int, c: Char ->
+                etArray[index].setText(c.toString())
+            }
+        }
+
+    init {
+        View.inflate(context, R.layout.view_otp, this).apply {
+            ButterKnife.bind(this@OtpView)
+        }
+
+        //adding
+        etArray[0].addTextChangedListener(GenericTextWatcher(etArray[0], etArray[1]))
+        etArray[1].addTextChangedListener(GenericTextWatcher(etArray[1], etArray[2]))
+        etArray[2].addTextChangedListener(GenericTextWatcher(etArray[2], etArray[3]))
+        etArray[3].addTextChangedListener(GenericTextWatcher(etArray[3], etArray[4]))
+        etArray[4].addTextChangedListener(GenericTextWatcher(etArray[4], etArray[5]))
+        etArray[5].addTextChangedListener(GenericTextWatcher(etArray[5], null))
+
+        //deleting
+        etArray[0].setOnKeyListener(GenericKeyEvent(etArray[0], null))
+        etArray[1].setOnKeyListener(GenericKeyEvent(etArray[1], etArray[0]))
+        etArray[2].setOnKeyListener(GenericKeyEvent(etArray[2], etArray[1]))
+        etArray[3].setOnKeyListener(GenericKeyEvent(etArray[3], etArray[2]))
+        etArray[4].setOnKeyListener(GenericKeyEvent(etArray[4], etArray[3]))
+        etArray[5].setOnKeyListener(GenericKeyEvent(etArray[5], etArray[4]))
+    }
+
+
+    private fun validateOtp(): Boolean {
+        for (et in etArray) {
+            if (et.text.isNullOrBlank()) {
+                return false
+            }
+        }
+        return true
+    }
+
+    companion object {
+        const val length = 6
+    }
+
+}
+
+class GenericKeyEvent internal constructor(private val currentView: EditText, private val previousView: EditText?) : View.OnKeyListener {
+    override fun onKey(p0: View?, keyCode: Int, event: KeyEvent?): Boolean {
+        if (event!!.action == KeyEvent.ACTION_DOWN && keyCode == KeyEvent.KEYCODE_DEL && currentView.id != R.id.otp1 && currentView.text.isEmpty()) {
+            previousView!!.text = null
+            previousView.requestFocus()
+            return true
+        }
+        return false
+    }
+
+
+}
+
+class GenericTextWatcher internal constructor(private val currentView: View, private val nextView: View?) : TextWatcher {
+    override fun afterTextChanged(editable: Editable) {
+        val text = editable.toString()
+        when (currentView.id) {
+            R.id.otp1 -> if (text.length == 1) nextView!!.requestFocus()
```

Comment:
All the code is same, just **R.id.otp6** case is different. But in that case, **nextView** is null anyways so you could rewrite the logic to 1 line only